### ISSUE TITLE
Dependencies not provided in tests.

### ISSUE THREAD

**Issue by Kotlinovsky** [google/dagger#1956]
Hi! I tried to use Hilt, and had a problem during testing

![image](https://user-images.githubusercontent.com/58388003/85874231-fb97bb00-b7da-11ea-95d6-5c27ed286f04.png)
![image](https://user-images.githubusercontent.com/58388003/85874243-00f50580-b7db-11ea-90e7-4db9f916d24a.png)
![image](https://user-images.githubusercontent.com/58388003/85874249-02bec900-b7db-11ea-927e-d7d9bb7c4e70.png)
![image](https://user-images.githubusercontent.com/58388003/85874254-05212300-b7db-11ea-8255-58186163b3fa.png)
![image](https://user-images.githubusercontent.com/58388003/85874262-081c1380-b7db-11ea-9f64-7d8f9a73f56b.png)

APPLICATION and COUNTRIES-FEATURE modules contain in INJECTOR Gradle-module.

**Comment by bcorso** [comment]
Hi Maxim, what was the actual error you ran into? Can you post it here?

**Comment by Kotlinovsky** [comment]
> Hi Maxim, what was the actual error you ran into? Can you post it here?

Dependencies injected to test class by rule.inject(), but not injected in View-class.

**Comment by bcorso** [comment]
The activity that hosts the view also needs to be annotated with `@AndroidEntryPoint`. An easy way to do this is to create a test activity nested in your test:

```java
  @AndroidEntryPoint
  public static final class TestActivity extends AppCompatActivity {}

  ...

  controller = Robolectric.buildActivity(TestActivity.class);
```

**Comment by Kotlinovsky** [comment]
![image](https://user-images.githubusercontent.com/58388003/85880925-7b2a8780-b7e5-11ea-8336-51b418cdcd56.png)
It's not working ;(

**Comment by Kotlinovsky** [comment]
![image](https://user-images.githubusercontent.com/58388003/85881204-ed02d100-b7e5-11ea-8d30-a461956cd637.png)
It's also not fixing my trouble.

**Comment by Kotlinovsky** [comment]
I checked instrumental tests (tests on device) with Hilt and dependencies successfully injects. But on local tests injection not working.

**Comment by bcorso** [comment]
Yep, I was able to reproduce your issue. 

Turns out our bytecode injection does not run over the test itself when running the test from AndroidStudio. There's a couple work arounds:

1. Use the long-form notation for any `@AndroidEntryPoint` usages in the test. So the previous example I gave would need to be:

```java
    @AndroidEntryPoint(AppCompatActivity.class)
    public static final class TestActivity extends Hilt_PhoneEditTextTest_TestActivity {}

    ...

    controller = Robolectric.buildActivity(TestActivity.class);
```

2. Run the test from the terminal using Gradle rather than AndroidStudio, e.g. `./gradlew test`

Thanks for reporting this!

We should make a note of this in the testing documentation.

**Comment by Kotlinovsky** [comment]
Thanks for supporting!

**Comment by danysantiago** [comment]
I wanted to add, that this is a known issue we are working with the Android Gradle Plugin team to fix, specifically [b/37076369](https://issuetracker.google.com/37076369).

The bytecode transform is done but when running tests from Android Studio the IDE does not grab the transformed classes, which causes injection not to occur in `@AndroidEntryPoint`-annotated classes in the test sources and when running tests from Android Studio (via the play button in the test method or class). Note that this only occurs for host-side tests, i.e. Robolectric not instrumentation tests. Meanwhile, if executed from command line it should work.

Alternatively you can configure Android Studio to execute tests via the Gradle test task instead of its internal task.
* Create a new 'Run Configuration' of type 'Gradle'
* Gradle project should be the Gradle module where the tests are.
* Task is usually `testDebug`
* Arguments should be `--tests MyTestClass`

Here is an example:
<img width="1104" alt="Screen Shot 2020-06-26 at 11 13 42 AM" src="https://user-images.githubusercontent.com/1844995/85890313-f66a4a00-b7a1-11ea-9918-9bc0b1103157.png">

We'll update the docs to mention the workarounds while a fix is being worked on.

**Comment by otaviosgoncalves** [comment]
This bug still remains....
